src/ert/run_models/everest_run_model.py

```diff
@@ -948,6 +1013,16 @@ def add(
             ),
         )
 
+    def find(
```

Comment:
This will return the first realization, that has the right control values. However, the cached is indexed by realization, because there are multiple realizations with the same controls. So, you probably need to pass the realization you want and first index into the cache before searching. Actually `get` below does that.